WooCommerce/WooCommerceTests/Yosemite/InAppPurchaseStoreTests.swift

```diff
@@ -176,35 +176,43 @@ final class InAppPurchaseStoreTests: XCTestCase {
     }
 
     func test_user_is_entitled_to_product_returns_false_when_not_entitled() throws {
-        // Given
-
         // When
-        let result = waitFor { promise in
-            let action = InAppPurchaseAction.userIsEntitledToProduct(productID: self.sampleProductID) { result in
-                    promise(result)
-                }
-            self.store.onAction(action)
+        Task {
+            do {
+                let isEntitled = try await userIsEntitledToProduct()
+                // Then
+                XCTAssertFalse(isEntitled)
+            } catch {
+                XCTAssert(error is WordPressApiError)
+            }
         }
-
-        // Then
-        let isEntitled = try XCTUnwrap(result.get())
-        XCTAssertFalse(isEntitled)
     }
 
     func test_user_is_entitled_to_product_returns_true_when_entitled() throws {
         // Given
         try storeKitSession.buyProduct(productIdentifier: sampleProductID)
 
         // When
-        let result = waitFor { promise in
-            let action = InAppPurchaseAction.userIsEntitledToProduct(productID: self.sampleProductID) { result in
-                    promise(result)
-                }
-            self.store.onAction(action)
+        Task {
+            do {
+                let isEntitled = try await userIsEntitledToProduct()
+                // Then
+                XCTAssertTrue(isEntitled)
+            } catch {
+                XCTAssert(error is WordPressApiError)
+            }
         }
+    }
+}
 
-        // Then
-        let isEntitled = try XCTUnwrap(result.get())
-        XCTAssertTrue(isEntitled)
+private extension InAppPurchaseStoreTests {
+    // Returns whether the user is entitled the product identified with the passed ID
+    func userIsEntitledToProduct() async throws -> Bool {
+        await withCheckedContinuation { (continuation: CheckedContinuation<Bool, Never>) in
+            let action = InAppPurchaseAction.userIsEntitledToProduct(productID: self.sampleProductID) { _ in
+                continuation.resume(returning: (true))
```

Comment:
I don't understand why we always return `true` here. Should we get the `Bool` value from the result and return it instead?

Reply:
Oh! Yes, much better to keep the same return type and use `Result`. Fixed on https://github.com/woocommerce/woocommerce-ios/pull/8462/commits/ffb542aff0d72d3f9d3d41fc88e2bbd3d7fd4d32